File: dg-cinema-infrastructure/src/main/java/devgraft/dgcinema/infrastructure/service/query/MovieSearchService.java
package devgraft.dgcinema.infrastructure.service.query;

import devgraft.dgcinema.domain.model.Movie;
import devgraft.dgcinema.domain.model.MovieNotFoundException;
import devgraft.dgcinema.domain.port.in.MovieSearchUseCase;
import devgraft.dgcinema.domain.port.out.MoviePort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@RequiredArgsConstructor
@Service
class MovieSearchService implements MovieSearchUseCase {
    private final MoviePort moviePort;

    @Override
    public Movie getMovie(final Long movieId) {
        return moviePort.findById(movieId)
                .orElseThrow(MovieNotFoundException::new);
    }

    @Override
    public List<Movie> getMovieList() {
        return moviePort.findAll();
    }
}

File: dg-cinema-infrastructure/src/main/kotlin/devgraft/dgcinema/service/MovieSearchService.kt
package devgraft.dgcinema.service

import devgraft.dgcinema.domain.model.Movie
import devgraft.dgcinema.domain.ports.`in`.MovieSearchUseCase
import devgraft.dgcinema.domain.ports.out.MoviePort
import org.springframework.stereotype.Service

@Service
internal class MovieSearchService(private val moviePort: MoviePort) : MovieSearchUseCase {
    override fun getMovie(movieId: Long): Movie {
        return moviePort.findById(movieId)
            .orElseThrow { RuntimeException("Movie with id $movieId not found") }
    }

    override fun getMovieList(): List<Movie> {
        return moviePort.findAll()
    }
}
